### DISCUSSION TITLE
Adding of two edge index

### DISCUSSION THREAD

**Discussion by shrimonmuke0202** [pyg-team/pytorch_geometric#2493] [Show and tell]
How can I add two edge_index and feed this new edge_index in the GCN layer?

How can I add my original edge_index with another sparse matrix which is generated from random gaussian distribution?

**Comment by adhikarirsr** [comment]
After adding the sparse random matrix, will you get a weighted adjacency matrix?

**Reply by shrimonmuke0202** [reply]
I want to add random Gaussian noise in the edge index ie in the adjacency matrix of a graph.

**Reply by adhikarirsr** [reply]
Navie way may look like this: 
You can convert your `edge_index` to Adjacency matrix using [`to_scipy_sparse_matrix`](https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html?highlight=to_scipy_sparse_matrix#torch_geometric.utils.to_scipy_sparse_matrix) and then add your sparse random Gaussian matrix to it. If I understood correctly then it will give you weighted adjacency matrix. Then you can convert that into edge index as follows:
```
row, col = adj.nonzero()
edge_index = torch.stack([row, col], dim=0)
edge_weight = adj[row, col]
```
`edge_weight` will contain the weight of the edges.

**Reply by rusty1s** [reply]
That is correct. If you are looking for a sparse version, you can also sample a fixed amount of edges via `torch_geometric.utils.negative_sampling`, create corresponding edge weights via `torch.randn(num_additional_edges)`, and concatenate both tensors to your original sparse adjacency matrix.
